packages/eas-cli/src/commands/build/cancel.ts

```diff
@@ -0,0 +1,129 @@
+import { Command } from '@oclif/command';
+import chalk from 'chalk';
+import gql from 'graphql-tag';
+import ora from 'ora';
+
+import { platformEmojis } from '../../build/constants';
+import { Platform } from '../../build/types';
+import { graphqlClient, withErrorHandlingAsync } from '../../graphql/client';
+import { AppPlatform, Build, BuildStatus } from '../../graphql/generated';
+import { BuildQuery } from '../../graphql/queries/BuildQuery';
+import log from '../../log';
+import {
+  findProjectRootAsync,
+  getProjectFullNameAsync,
+  getProjectIdAsync,
+} from '../../project/projectUtils';
+import { confirmAsync, selectAsync } from '../../prompts';
+
+async function cancelBuildAsync(buildId: string): Promise<Pick<Build, 'id' | 'status'>> {
+  const data = await withErrorHandlingAsync(
+    graphqlClient
+      .mutation<{ build: { cancel: { id: string; status: BuildStatus } } }>(
+        gql`
+          mutation CancelBuildMutation($buildId: ID!) {
+            build(buildId: $buildId) {
+              cancel {
+                id
+                status
+              }
+            }
+          }
+        `,
+        { buildId }
+      )
+      .toPromise()
+  );
+  return data.build!.cancel;
+}
+
+const appPlatformMap = {
+  [AppPlatform.Android]: Platform.Android,
+  [AppPlatform.Ios]: Platform.iOS,
+};
+
+function formatUnfinishedBuild(
+  build: Pick<Build, 'id' | 'platform' | 'status' | 'createdAt'>
+): string {
+  const platform = platformEmojis[appPlatformMap[build.platform]];
+  const startTime = new Date(build.createdAt).toLocaleString();
+  const status = chalk.blue(build.status === BuildStatus.InQueue ? 'in-queue' : 'in-progress');
+  return `${platform} Started at: ${startTime}, Status: ${status}, Id: ${build.id}`;
+}
+
+async function selectBuildToCancelAsync(
+  projectId: string,
+  fullProjectName: string
+): Promise<string | null> {
+  const spinner = ora().start('Fetching the unfinished builds…');
+  let builds;
+  try {
+    const [inQueueBuilds, inProgressBuilds] = await Promise.all([
+      BuildQuery.allForAppAsync(projectId, { status: BuildStatus.InQueue }),
+      BuildQuery.allForAppAsync(projectId, { status: BuildStatus.InProgress }),
+    ]);
+    spinner.stop();
+    builds = [...inQueueBuilds, ...inProgressBuilds];
+  } catch (error) {
+    spinner.fail(
+      `Something went wrong and we couldn't fetch the builds for the project ${fullProjectName}.`
+    );
+    throw error;
+  }
+  if (builds.length === 0) {
+    log.warn(`There are no unfinished builds for the project ${fullProjectName}.`);
```

Comment:
I'd prefer to move this log outside of this function (to `if (!buildId) { ... }`).
`BuildCancel.run` is the main function and code like `if (!buildId} { return; }` doesn't say anything. I had to check the implementation of `selectBuildToCancelAsync` to see what's the case if there aren't any uncompleted builds.

Reply:
```suggestion
    log.warn(`There aren't any uncompleted builds for project ${fullProjectName}.`);
```